.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,237 @@
+      - name: Run All Docker Containers - Sample App + OTel Collector + Load Generator + Alarm Poller
+        id: check-failure-during-performance-tests
+        continue-on-error: true
+        working-directory: .github/docker-performance-tests
+        env:
+          INSTANCE_ID: ${{ github.run_id }}-${{ github.run_number }}
+          LOG_GROUP_NAME: otel-sdk-performance-tests
+          # Also uses:
+          # AWS_ACCESS_KEY_ID
+          # AWS_SECRET_ACCESS_KEY
+          # AWS_SESSION_TOKEN
+          # APP_PATH
+          # TARGET_SHA
+          # LISTEN_ADDRESS_PORT
+          # LOGS_NAMESPACE
+          # APP_PROCESS_COMMAND_LINE_DIMENSION_VALUE
+          # APP_PROCESS_EXECUTABLE_NAME
+          # HOSTMETRICS_INTERVAL_SECS
+          # TEST_DURATION_MINUTES
+          # NUM_OF_CPUS
+          # CPU_LOAD_THRESHOLD
+          # TOTAL_MEMORY_THRESHOLD
+          # AWS_DEFAULT_REGION
+          # MATRIX_COMMIT_COMBO
+          # GITHUB_RUN_ID
+        run: |-
+          docker-compose up --build;
+          RUN_TESTS_EXIT_CODE=$(
+            docker inspect $(
+              docker ps --quiet --all --filter "name=docker-performance-tests_alarms-poller"
+            ) --format="{{.State.ExitCode}}"
+          );
+          echo "RUN_TESTS_EXIT_CODE=$RUN_TESTS_EXIT_CODE" | tee --append $GITHUB_ENV;
+          exit $RUN_TESTS_EXIT_CODE;
+      - name: Fail early if Soak Tests failed to start
+        if: ${{ env.RUN_TESTS_EXIT_CODE == '' || env.RUN_TESTS_EXIT_CODE == 1 }}
+        run: exit 1;
+
+    # MARK: - Report on Performance Test Results
+
+      - name: Install script dependencies
+        run: pip install boto3
+      - name: Get a snapshot of metrics and commit them to the repository
+        run: |
+          python3 .github/scripts/performance-tests/produce_metric_widget_images.py \
+            --logs-namespace ${{ env.LOGS_NAMESPACE }} \
+            --metrics-period ${{ env.HOSTMETRICS_INTERVAL_SECS }} \
+            --num-of-cpus ${{ env.NUM_OF_CPUS }} \
+            --app-process-command-line-dimension-value "${{ env.APP_PROCESS_COMMAND_LINE_DIMENSION_VALUE }}" \
+            --target-sha ${{ env.TARGET_SHA }} \
+            --github-run-id ${GITHUB_RUN_ID} \
+            --test-duration-minutes ${{ env.TEST_DURATION_MINUTES }} \
+            --cpu-load-threshold ${{ env.CPU_LOAD_THRESHOLD }} \
+            --total-memory-threshold ${{ env.TOTAL_MEMORY_THRESHOLD }} \
+            --app-platform ${{ matrix.app-platform }} \
+            --instrumentation-type ${{ matrix.instrumentation-type }} \
+            --max-benchmarks-to-keep ${{ env.MAX_BENCHMARKS_TO_KEEP }} \
+            --github-repository ${GITHUB_REPOSITORY}
+          echo "::warning::Checkout Snapshots at this link: https://github.com/${GITHUB_REPOSITORY}/blob/gh-pages/soak-tests/snapshots/commits/${{ env.TARGET_SHA }}/runs/${GITHUB_RUN_ID}/${{ matrix.app-platform }}";
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com";
```

Comment:
Is there a significance to the email?

Reply:
If its not important than possibly use [example.com](https://datatracker.ietf.org/doc/html/rfc6761#section-6.5) instead

Reply:
I've seen github-actions@github.com and actions@github.com used. It's not a big deal though. Was just curious about it.

Reply:
Sorry I didn't mean it's not important. Take the example of this issue in ADOT Java:

You can see that it has the logo for the GitHub Actions bot. This is valuable because it signals to automation effort we put into this step. You can see that GitHub bot has the e-mail I have here and matches the ID lookup in this API: https://api.github.com/users/github-actions%5Bbot%5D. That's where I got this e-mail. It's also mentioned in this GitHub Forum post: https://github.community/t/github-actions-bot-email-address/17204/6

We will lose the GitHub actions bot profile picture/naming/etc if we change the e-mail.

Reply:
Having the icon and name _is_ something magic about this number that I reasoned about 6 months ago when I wrote this action 😝.

> I've seen [github-actions@github.com](mailto:github-actions@github.com) and [actions@github.com](mailto:actions@github.com) used. It's not a big deal though. Was just curious about it.

Cool! I'd be interested to know what the user looks like when those e-mails are used 🙂 Will keep it in mind!